model/modelpb/client.pb.json.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package modelpb
+
+import (
+	"net/netip"
+
+	"github.com/elastic/apm-data/model/internal/modeljson"
+)
+
+func (c *Client) toModelJSON(out *modeljson.Client) {
+	*out = modeljson.Client{
+		Domain: c.Domain,
+		Port:   int(c.Port),
+	}
+	if c.Ip != "" {
+		if _, err := netip.ParseAddr(c.Ip); err == nil {
+			out.IP = c.Ip
+		}
+	}
```

Comment:
It doesn't seem ideal that we need to parse the IP here, to convert it into a netip.Addr, only to then convert it back into a string.

I wonder if we should introduce a more efficient IP type in our protobuf, along these lines: https://stackoverflow.com/questions/51395679/what-is-the-most-efficient-protobuf-type-in-c-for-storing-ipv4-or-ipv6-addre

WDYT? Is it overkill?

Reply:
No, I fully agree we should store it in a better way (bytes ?) but probably in a followup PR (I think there's a TODO about it)

This goal of `ParseAddr` here is to replicate the `ip.isValid()` call in the client model. Looking more at it, it seems that that's used to make sure that the ip is not a zero value so the if check might with the empty string might be enough here. WDYT ?

Reply:
It's fine as it is for now, but a followup would be good before we commit to keeping these protobufs stable.

Reply:
Opened https://github.com/elastic/apm-data/issues/47